### ISSUE TITLE
Error in angular after build for version 1.4.1 

### ISSUE THREAD

**Issue by acharyaks90** [appvision-gmbh/json2typescript#133]
After making build and serving for version 1.4.1
Uncaught Error: Fatal error in JsonConvert. It is mandatory to pass a string as parameter in the @JsonObject decorator.

Use @JsonObject(classId) where classId is a string.

    at RCAU.t.JsonObjec

Expected no error.

Note: work fine with version 1.3.0

**Comment by andreas-aeschlimann** [comment]
Hello and thanks for your question. This is not a bug, this is actually a feature.

You must provide a class identifier in the decorator now. Check the docs for more info.

It is true that version < 1.4 does not force you to use the parameter. You could downgrade to 1.2.5 or 1.3.0. But if you do that, you risk to get problems in your applications. The reason is a pull request (https://github.com/AppVision-GmbH/json2typescript/pull/121), it changed the way of detecting properties of child classes.

TLDR: If you use class inheritance (extend classes), you will run into bugs if you don't provide the first parameter in the decorator. Version 1.4 helps you with that because it will throw an error if you forget it.

**Comment by ericgruss** [comment]
@andreas-aeschlimann breaking changes should change major version number always.
Given a version number MAJOR.MINOR.PATCH, increment the:

MAJOR version when you make incompatible API changes,
MINOR version when you add functionality in a backwards compatible manner, and
PATCH version when you make backwards compatible bug fixes.
Additional labels for pre-release and build metadata are available as extensions to the MAJOR.MINOR.PATCH format.

**Comment by andreas-aeschlimann** [comment]
Already commented on this in #134 about the thoughts I put into the current update. 

TLDR: I wanted developers to run into the error because otherwise they would never notice that they might implement a bug.

On another note, this library has never followed SEMVER, but I agree it is a good idea to start it with the next bigger improvements.
